test/type/explicit_existential.swift

```diff
@@ -478,6 +478,22 @@ func testAnyFixIt() {
   let _: (((((~Copyable) & NonCopyableHasAssoc) & NonCopyableHasAssoc).Type.Type)).Type?
   let _: any (((((~Copyable) & NonCopyableHasAssoc) & NonCopyableHasAssoc).Type.Type)).Type // OK
 
+  // Misplaced '!'.
+  // expected-error@+1 {{force unwrapped 'any' type must be written (any HasAssoc)!}}{{10-23=(any HasAssoc)!}}
+  let _: any HasAssoc!
+
+  // expected-error@+1 {{using '!' is not allowed here; perhaps '?' was intended}}{{16-29=(any HasAssoc)?}}
+  let _: (Int, any HasAssoc!)
+
+  // expected-error@+1 {{using '!' is not allowed here; perhaps '?' was intended}}{{12-19=(any PP)?}}
+  let _: G<any PP!>
+
+  // expected-error@+1 {{using '!' is not allowed here; perhaps '?' was intended}}{{11-18=(any PP)?}}
+  let _: (any PP!) -> Void
+
+  // expected-error@+1 {{using '!' is not allowed here; perhaps '?' was intended}}{{11-17=(any PP)?}}
+  let _: (any P!)?
+
```

Comment:
@AnthonyLatsis now i'm matching more cases.

because i renamed the `incorrect_optional_any` i ended up breaking the diagnostics of `let _ = any P?`

but i think it's easy to fix, by emitting a special (the old message) for this case